### PR TITLE
@damassi => FileInput component

### PR DESCRIPTION
**Pull request by eessex**
Closed the earlier PR today in favor of this one - which actually includes js tests. 
New component for use all around the content2 app, not yet implemented here.

- Adds component FileInput for uploading with gemup
- Adds js to mocha compiling
- Adds `*.test.js` to makefile test command
- Adds enzyme, jest, and jest-enzyme
- Updates reaction
- Adds done to Articles Retrieval test (hopefully fixing circle issues)

### CODE REVIEW

File: client/components/file_input/test/index.test.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    const container = wrapper.find('.file-input')
+    expect(container.text()).toMatch(/Drag & Drop or Click to Upload/)
+  })
+
+  it('renders the default size limit', () => {
+    const wrapper = mount(
+      <FileInput />
+    );
+    const container = wrapper.find('.file-input')
+    expect(container.text()).toMatch(/Up to 30mb/)
```
Comment (damassi) [inline client/components/file_input/test/index.test.js]: Just an FYI, `toMatch` also accepts strings

File: client/components/file_input/test/index.test.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+  it('calls upload when a file is selected', () => {
+    const spy = jest.fn()
+    FileInput.prototype.uploadFile = (e) => { spy(e) }
+    const wrapper = mount(
+      <FileInput />
+    );
+    const file = new Blob([], {type : 'img/jpg', name: 'name'})
+    wrapper.find('input').simulate('change', {target: {files: [file]}})
+    expect(spy).toHaveBeenCalled()
```
Comment (eessex) [inline client/components/file_input/test/index.test.js]: Would like to mock uploadFiles' gemup function/callbacks here but haven't yet cracked mocking dependencies in jest.

Comment (damassi) [inline client/components/file_input/test/index.test.js]: I'm a huge fan of `babel-plugin-rewire` for mocks (much simpler / cleaner IMO than Jest) but from what I understand Jest's mocking works well; here's an example: https://github.com/artsy/prediction/blob/master/src/components/LotEventList.test.js#L1. But I could be doing this totally wrong!

**Comment by damassi** [comment]
👍

**Pull request merged**